146 openfoam example

MarcDuQuesne opened the pull request:
WIP

I added the model, cfg and observations for a simple openfoam/openda run using the EnKF engine. 

TODO: 
add the DUD engine cfg files.

aljavrieling committed:
Moved Role to IExchangeItem and added PrevRole en getPrevRole to IPrevExchangeItem + refactoring

aljavrieling committed:
Moved Role to IExchangeItem and added PrevRole en getPrevRole to IPrevExchangeItem + refactoring

aljavrieling committed:
removed model_delft3d from build.xml

aljavrieling committed:
removed model_delft3d from build.xml

aljavrieling committed:
eliminated all direct implementations of IPrevExchangeItem

aljavrieling committed:
eliminated all direct implementations of IPrevExchangeItem

erikpelgrim committed:
Initial code model external socket

erikpelgrim committed:
Add config reader test case

erikpelgrim committed:
Add config reader test case

erikpelgrim committed:
Use model results as observed values

erikpelgrim committed:
Move external socket code to model instance

erikpelgrim committed:
Implement compute for External socket model

erikpelgrim committed:
Implement sendFinalParameters for External socket model

erikpelgrim committed:
Add model_external_socket to build.xml

erikpelgrim committed:
Rename Exchange Items to ExternalSocketParameter & ExternalSocketResult

erikpelgrim committed:
Lookup local host address instead of hardcoded 127.0.0.1

erikpelgrim committed:
Add logging for debugging non responsive Socket

erikpelgrim committed:
Add wait for socket input stream

erikpelgrim committed:
Use DataInputStream to prevent blocking

erikpelgrim committed:
Do not wait for response after sending final parameters

erikpelgrim committed:
Add ExternalSocketSimplexAlgorithm

erikpelgrim committed:
Add ExternalSocketSCEAlgorithm

erikpelgrim committed:
Add ExternalSocketPowellAlgorithm

erikpelgrim committed:
Add ExternalSocketGriddedFullSearchAlgorithm

erikpelgrim committed:
Add ExternalSocketConjugateGradientAlgorithm and ExternalSocketConjugateGLUEAlgorithm

erikpelgrim committed:
Add ExternalSocketBFGSAlgorithm

erikpelgrim committed:
Support flexible number of model results

aljavrieling committed:
Revert "removed model_delft3d from build.xml"

This reverts commit 97ad26dc16482c6c29a486c7983351bec5f26fc6.

aljavrieling committed:
Revert "removed model_delft3d from build.xml"

This reverts commit 97ad26dc16482c6c29a486c7983351bec5f26fc6.

aljavrieling committed:
eliminated implements IPrevExchangeItem in model_delft3d

aljavrieling committed:
eliminated implements IPrevExchangeItem in model_delft3d

aljavrieling committed:
eliminated getValueType, replaced by method getValuesType of IExchangeItem

aljavrieling committed:
eliminated getValueType, replaced by method getValuesType of IExchangeItem

aljavrieling committed:
moved remaining methods from IPrevExchangeItem to IExchangeItem and refactoring. No beautifying.

aljavrieling committed:
moved remaining methods from IPrevExchangeItem to IExchangeItem and refactoring. No beautifying.

aljavrieling committed:
removed unused import statements and last occurences of IPrevExchangeItem

aljavrieling committed:
removed unused import statements and last occurences of IPrevExchangeItem

erikpelgrim committed:
Speed up socket client waiting for data

erikpelgrim committed:
Add sending optional standard deviation to final parameters

theavuik committed:
Replaced IoObjectInterface by IDataObject for D3dField2Dfile

theavuik committed:
Replaced IoObjectInterface by IDataObject for D3dField2Dfile

theavuik committed:
Removed getExchangeItems() from D3dField2Dfile

theavuik committed:
Removed getExchangeItems() from D3dField2Dfile

theavuik committed:
Add @Override to D3dField2Dfile functions initialize and finish

theavuik committed:
Add @Override to D3dField2Dfile functions initialize and finish

theavuik committed:
Replaced IoObjectInterface by IDataObject for D3dAstroComponentFiles

theavuik committed:
Replaced IoObjectInterface by IDataObject for D3dAstroComponentFiles

wkramer committed:
Add comments & style improvements

wkramer committed:
Add comments & style improvements

theavuik committed:
Added more imports to reduce Codacy issues

theavuik committed:
Added more imports to reduce Codacy issues

wkramer committed:
Merge branch '107-eliminate-IPrevExchangeItem' of https://github.com/OpenDA-Association/OpenDA into 107-eliminate-IPrevExchangeItem

theavuik committed:
Implemented AbstractDataObject and used this class in D3dRoughParamsFile

theavuik committed:
Implemented AbstractDataObject and used this class in D3dRoughParamsFile

theavuik committed:
Merge remote-tracking branch 'origin/107-eliminate-IPrevExchangeItem' into 107-eliminate-IPrevExchangeItem

theavuik committed:
Replaced IoObjectInterface by extension of AbstractDataObject for DummyAstroIoObject

theavuik committed:
Replaced IoObjectInterface by extension of AbstractDataObject for DummyAstroIoObject

theavuik committed:
Removed unused class DummyIoObject

theavuik committed:
Removed unused class DummyIoObject

theavuik committed:
Replaced IoObjectInterface by extension of AbstractDataObject for DummyParametersIoObject

theavuik committed:
Replaced IoObjectInterface by extension of AbstractDataObject for DummyParametersIoObject

theavuik committed:
Reduce Codacy errors

theavuik committed:
Reduce Codacy errors

theavuik committed:
Use of AbstractDataObject for DummySeries

theavuik committed:
Use of AbstractDataObject for DummySeries

theavuik committed:
Use of AbstractDataObject for DummyTimeInfoIoObject

theavuik committed:
Use of AbstractDataObject for DummyTimeInfoIoObject

theavuik committed:
Removed unused imports

theavuik committed:
Removed unused imports

theavuik committed:
Update for DummyState

theavuik committed:
Update for DummyState

theavuik committed:
Implementation of getExchangeItemIDs for roles.

theavuik committed:
Implementation of getExchangeItemIDs for roles.

theavuik committed:
Improve codacy log of AbstractDataObject

theavuik committed:
Improve codacy log of AbstractDataObject

theavuik committed:
Update of test since the number of IOObjectInterfaces reduces

theavuik committed:
Update of test since the number of IOObjectInterfaces reduces

theavuik committed:
Use of AbstractDataObject for EfdcEventTox2InploObject

theavuik committed:
Use of AbstractDataObject for EfdcEventTox2InploObject

theavuik committed:
Reduce codacy errors

theavuik committed:
Reduce codacy errors

theavuik committed:
Use of AbstractDataObject for EfdcGridTimeSeriesIoObject

theavuik committed:
Use of AbstractDataObject for EfdcGridTimeSeriesIoObject

theavuik committed:
UIModel: remove unused imports

theavuik committed:
UIModel: remove unused imports

theavuik committed:
Removed unused DummySeries class

theavuik committed:
Removed unused DummySeries class

theavuik committed:
Use of AbstractDataObject for EfdcInpIoObject

theavuik committed:
Use of AbstractDataObject for EfdcInpIoObject

theavuik committed:
Use of AbstractDataObject for EfdcRestartFileIoObject

theavuik committed:
Use of AbstractDataObject for EfdcRestartFileIoObject

theavuik committed:
AbstractDataObject: added comments

theavuik committed:
AbstractDataObject: added comments

theavuik committed:
EfdcTimeSeriesIoObject: used AbstractDataObject

theavuik committed:
EfdcTimeSeriesIoObject: used AbstractDataObject

theavuik committed:
ForcingWrapper: used AbstractDataObject

theavuik committed:
ForcingWrapper: used AbstractDataObject

theavuik committed:
Reduce IntelliJ yellow remarks

theavuik committed:
Reduce IntelliJ yellow remarks

theavuik committed:
Reduce IntelliJ yellow remarks - part 2

theavuik committed:
Reduce IntelliJ yellow remarks - part 2

theavuik committed:
Undo Deprecated statement in UciIoObject for now

theavuik committed:
Undo Deprecated statement in UciIoObject for now

theavuik committed:
SimpleBbAsciiFile: use AbstractDataobject

theavuik committed:
SimpleBbAsciiFile: use AbstractDataobject

theavuik committed:
Check if the ids from getExchangeItemIDs equal those from exchangeItem.getId()

theavuik committed:
Check if the ids from getExchangeItemIDs equal those from exchangeItem.getId()

theavuik committed:
Small improvements

theavuik committed:
Small improvements

theavuik committed:
Update of InitialStatesWrapper

theavuik committed:
Update of InitialStatesWrapper

theavuik committed:
NoosDataObject: updated to use AbstractDataObject, new noos files, and more tests (similar to those from IoObject case)

theavuik committed:
NoosDataObject: updated to use AbstractDataObject, new noos files, and more tests (similar to those from IoObject case)

theavuik committed:
ParametersWrapper: use of AbstractDataObject

theavuik committed:
ParametersWrapper: use of AbstractDataObject

theavuik committed:
Fix for noos files in dataobject tests

theavuik committed:
Fix for noos files in dataobject tests

theavuik committed:
Use dates instead of mjd in NoosDataObjectTest

theavuik committed:
Use dates instead of mjd in NoosDataObjectTest

theavuik committed:
AbstractDataObject for ResultFileWrapper

theavuik committed:
AbstractDataObject for ResultFileWrapper

theavuik committed:
AbstractDataObject for TimesWrapper

theavuik committed:
AbstractDataObject for TimesWrapper

theavuik committed:
AbstractDataObject for TreeVectorIoObject

theavuik committed:
AbstractDataObject for TreeVectorIoObject

theavuik committed:
Renaming to dataObject, and better objects

theavuik committed:
Renaming to dataObject, and better objects

theavuik committed:
Renaming to dataObject, and small updates

theavuik committed:
Renaming to dataObject, and small updates

theavuik committed:
AbstractDataObject for myWrapper

theavuik committed:
AbstractDataObject for myWrapper

MarcDuQuesne committed:
The model, the cfg and the observations.

MarcDuQuesne committed:
The model, the cfg and the observations.

MarcDuQuesne committed:
Added a readme with the versions of the software.

MarcDuQuesne committed:
Added a readme with the versions of the software.

wkramer commented:
Is there a sequential simulation configuration, and a ensemble sequential model configuration? That would be really helpful.

MarcDuQuesne commented:
@wkramer Yes, we also have a sequential version. Let me see if I can produce one.

theavuik committed:
ioObjects: test such that old ioobjects still work

theavuik committed:
ioObjects: test such that old ioobjects still work

wkramer commented:
`start_openfoam.sh` is missing

theavuik committed:
From ioObject to dataObject: update of xsd

theavuik committed:
From ioObject to dataObject: update of xsd

MarcDuQuesne committed:
Using the python format for the results instead of matlab.

MarcDuQuesne committed:
Using the python format for the results instead of matlab.

MarcDuQuesne committed:
Added the cfg files to use DUD instead of EnKF.

MarcDuQuesne committed:
Added the cfg files to use DUD instead of EnKF.

theavuik committed:
java: updates of files where ioObject was wrongly used

theavuik committed:
java: updates of files where ioObject was wrongly used

MarcDuQuesne commented:
you are right. It's because of one .gitignore. Let me add it and run it.

MarcDuQuesne committed:
Added file missing because of a .gitignore

MarcDuQuesne committed:
Added file missing because of a .gitignore

theavuik committed:
xmls changed such that dataobjects are used instead of ioobjects

theavuik committed:
xmls changed such that dataobjects are used instead of ioobjects

MarcDuQuesne committed:
Updated the model to a newer version.

MarcDuQuesne committed:
Updated the model to a newer version.

MarcDuQuesne committed:
Adapted the configuration files to the new model.

The heatsources are grouped. This is done through the use of the
openfoam #calc directive which in turn relies on the dynamic compilation
of some code through wmake.

MarcDuQuesne committed:
Adapted the configuration files to the new model.

The heatsources are grouped. This is done through the use of the
openfoam #calc directive which in turn relies on the dynamic compilation
of some code through wmake.

MarcDuQuesne commented:
@wkramer could you please test whether it all works now? 

When I run `oda_run.sh EnKF.oda` everything works for me, but I am using a rather old version of OpenDA (OpenDA  2.4.4.1 and OpenFOAM  v1712).

erikpelgrim committed:
Add dictionary for IoObjects renamed to DataObjects

erikpelgrim committed:
Add dictionary for IoObjects renamed to DataObjects

Commit:
Improved IoObject->DataObject backwards compatibility.

Commit:
Improved IoObject->DataObject backwards compatibility.

erikpelgrim committed:
Remove usage of BBStochModel

erikpelgrim committed:
Codacy fixes

erikpelgrim committed:
Codacy fixe

nilsvanvelzen committed:
Merge pull request #152 from OpenDA-Association/ProbabilisticToolKit

Probabilistic tool kit

wkramer committed:
Replace IPrevExchangeItem.Role

wkramer committed:
Replace IPrevExchangeItem.Role

wkramer committed:
Fix for allElementsFromDataObject

wkramer committed:
Fix for allElementsFromDataObject

wkramer committed:
Explicit Role import

wkramer committed:
Revert previous commit

theavuik committed:
Put CoreResources in modules instead of separate jar files from them

theavuik committed:
Put CoreResources in modules instead of separate jar files from them

theavuik committed:
Custom added to ValueTypes for ExchangeItems, and added todos for replacing D3Dresults with his files.

theavuik committed:
Custom added to ValueTypes for ExchangeItems, and added todos for replacing D3Dresults with his files.

Commit:
Fixed xml-parsing for dataObject and IoObject based stochObserver

Commit:
Fixed xml-parsing for dataObject and IoObject based stochObserver

Commit:
Improved logging of metaswap state file changes

Commit:
Merge branch '107-eliminate-IPrevExchangeItem' of https://github.com/OpenDA-Association/OpenDA into 107-eliminate-IPrevExchangeItem

erikpelgrim committed:
First attempt to fix travis for model_external_socket

erikpelgrim committed:
Second attempt to fix travis for model_external_socket

erikpelgrim committed:
Second attempt to fix travis

erikpelgrim committed:
Disable local test

erikpelgrim committed:
Comment out local test

erikpelgrim committed:
Comment out local test, enable extra debug logging

erikpelgrim committed:
Cleanup extra debug logging

nilsvanvelzen committed:
Merge pull request #153 from OpenDA-Association/FixBuildProblems

Fix build problems

erikpelgrim committed:
Update script for teamcity on the DPC according to suggestion of Tim van den Aardweg

aljavrieling committed:
Update script for Teamcity on DPC, Deltares issue tracker TEAMCITY-286

Commit:
Fixed compile errors that were a result of the removal of IPrevExchangeItem

nilsvanvelzen committed:
Merge pull request #155 from OpenDA-Association/FixTeamCityDPC

Fix team city dpc

theavuik committed:
Remove iPrevExchangeItem from java files

theavuik committed:
Remove model_metaswap from iml

theavuik committed:
Add more options to valueTypes in IExchangeItem

theavuik committed:
Remove 'required' for ioObjectId and dataObjectId in xsd

erikpelgrim committed:
Fix xsd's by adding deprecated ioObject element for backwards compatibility and rename dataStochObservers.xsd back to openDaStochObserver.xsd

Commit:
Merge pull request #129 from OpenDA-Association/107-eliminate-IPrevExchangeItem

WIP: 107 eliminate IPrevExchangeItem

waynezw0618 commented:
Hi @MarcDuQuesne 
It doesn't work for me here. since the mesh is not like what you are using. see here[https://github.com/OpenDA-Association/OpenDA/issues/146](url)

Commit:
Remove oracle jdk from travis test

nilsvanvelzen committed:
Merge pull request #160 from OpenDA-Association/travis-disable-oraclejdk

Remove oracle jdk from travis test

aljavrieling committed:
Merge branch '146-openfoam-example' of https://github.com/OpenDA-Association/OpenDA into 146-openfoam-example

aljavrieling committed:
Added README.md to explain the example, disclaimer for software versions.

xiaodog commented:
> WIP
> 
> I added the model, cfg and observations for a simple openfoam/openda run using the EnKF engine.
> 
> TODO: add the DUD engine cfg files.

Dear MarcDuQuesne,
 Could you please send the ' openfoam/openda' case to me, i can't find it in your link. thank u.
